hpx/components/containers/unordered/partition_unordered_map_component.hpp

```diff
@@ -0,0 +1,601 @@
+        }
+
+        /// Checks if the container has no elements, i.e. whether
+        /// begin() == end().
+        bool empty() const
+        {
+            return partition_unordered_map_.empty();
+        }
+
+        ///////////////////////////////////////////////////////////////////////
+        // Element access API's
+        ///////////////////////////////////////////////////////////////////////
+
+        /// Return the element at the position \a pos in the partition_unordered_map
+        /// container.
+        ///
+        /// \param pos Position of the element in the partition_unordered_map
+        ///
+        /// \return Return the value of the element at position represented
+        ///         by \a pos.
+        ///
+        struct erase_on_exit
+        {
+            erase_on_exit(data_type& m, typename data_type::iterator& it)
+              : m_(m), it_(it)
+            {}
+            ~erase_on_exit()
+            {
+                m_.erase(it_);
+            }
+
+            data_type& m_;
+            typename data_type::iterator& it_;
+        };
+
+        T get_value(Key const& key, bool erase)
+        {
+            typename data_type::iterator it = partition_unordered_map_.find(key);
+            if (it == partition_unordered_map_.end())
+            {
+                HPX_THROW_EXCEPTION(bad_parameter,
+                    "partition_unordered_map::get_value",
+                    "unable to find requested key in this partition of the "
+                    "unordered_map");
+            }
+
+            if (!erase)
+                return it->second;
+
+            erase_on_exit t(partition_unordered_map_, it);
+            return it->second;
+        }
+
+        /// Return the element at the position \a pos in the partition_unordered_map
+        /// container.
+        ///
+        /// \param pos Positions of the elements in the partition_unordered_map
+        ///
+        /// \return Return the values of the elements at position represented
+        ///         by \a pos.
+        ///
+        std::vector<T> get_values(std::vector<Key> const& keys)
+        {
+            std::vector<T> result;
+            result.reserve(keys.size());
+
+            for (std::size_t i = 0; i != keys.size(); ++i)
+                result.push_back(partition_unordered_map_[keys[i]]);
+
+            return result;
+        }
+
+        ///////////////////////////////////////////////////////////////////////
+        // Modifiers API's in server class
+        ///////////////////////////////////////////////////////////////////////
+
+        /// Copy the value of \a val in the element at position \a pos in the
+        /// partition_unordered_map container.
+        ///
+        /// \param pos   Position of the element in the partition_unordered_map
+        ///
+        /// \param val   The value to be copied
+        ///
+        void set_value(Key const& pos, T const& val)
+        {
```

Comment:
Not sure if this is possible. In order to calculate the sequence umber of the current partition we would need to know how many partitions are created overall. This information is not available to each of the partitions themselves (at least not now).